client/src/Options.js

```diff
@@ -57,66 +59,71 @@ JoinGame.propTypes = {
 };
 
 
-class Options extends Component {
-    static contextType = UserIdContext;
+function CreateGame(props) {
 
-    constructor (props) {
-      super(props);
-      this.state = {
-        start_game: false,
-        game_id: ''
-      }  ;
-      this.create_game = this.create_game.bind(this);
-      this.handleJoinGame = this.handleJoinGame.bind(this);
-    }
+  const {onCreateGame, onJoinGame} = props;
+  
+  const userId = useContext(UserIdContext);
 
-    create_game() {
-      axios.post('/create_game/' + this.context, {}).
-        then(response => this.handle_create_game_response(response), 
-          reason => this.handle_create_game_error(reason));
-    }
+  const handleCreateGameResponse = (response) => {
+    console.log(response);
+    var pin = response.data.game_id;
+    onCreateGame();
+    onJoinGame(pin);
+  };
 
-    handle_create_game_response(response) {
-      console.log(response);
-      this.setState({game_id: response.data.game_id});
-      this.setState({start_game: true});
+  const handleCreateClick = async () => {   
+    try {
+      const response = await axios.post( `/create_game/${userId}`, {});
+      handleCreateGameResponse(response);
+    } catch (error) {
+      //TODO: handle create game error;
+      console.log('Create game has thrown and exception');
     }
+  };
 
-    handle_create_game_error(reason) {
-      // TODO
-    }
+  return (
+    <div>
+      <button className='button' onClick={() => handleCreateClick()}>Create Game!</button>
+    </div>
+  );
 
-    handleJoinGame(input_game_id) {
-      this.setState({game_id: input_game_id});
-      this.setState({start_game: true});
-    }
+}
 
-    render_start_game() {
-      return (
-        <div className='main__container'>
-          <GamePlay game_id={this.state.game_id} />
-        </div>
-      );
-    }
+CreateGame.propTypes = {
+  onJoinGame: PropTypes.func.isRequired,
+  onCreateGame: PropTypes.func,
+};
 
-    render_regular () {
-      return (
-        <div className='main__container'>
-                User {window.$id} -  What Do You Want To Do? <br/> <br/>
-          <button className='button' onClick={this.create_game}> Create Game </button> <br/> <br/>
-          <JoinGame onJoinGame={this.handleJoinGame} />
-        </div>
-      );
-    }
+function Options() {
+  
+  // const userId = useContext(UserIdContext);
+  // const gameId = useContext(GameIdContext);
+  const [pinCode, setPicCodeField] = useState(-1);
```

Comment:
So the naming conventions is pinCode, SetPinCode (x, setx)
Also, usually you set initial value like this to undefined (which is commonly used in js to represent 'no value yet').
That way you can just do later: if (pinCode) {} else {}

```suggestion
  const [pinCode, setPicCode] = useState(undefined);
```